mailpoet/assets/js/src/automation/editor/components/filters/list.tsx

```diff
@@ -103,13 +107,15 @@ export function FiltersList(): JSX.Element | null {
                   </span>{' '}
                   <Value filter={filter} />
                 </div>
-                <Button
-                  className="mailpoet-automation-filters-list-item-remove"
-                  isSmall
-                  onClick={() => onDelete(step.id, filter)}
-                >
-                  <Icon icon={closeSmall} />
-                </Button>
+                {allowDelete && (
```

Comment:
Some logic like this is also needed for the operator radio boxes imo
![image](https://github.com/mailpoet/mailpoet/assets/6458412/06ce0de6-b571-4f6a-ac1d-92d990c435c7)